var/spack/repos/builtin/packages/lbann/package.py

```diff
@@ -51,8 +51,13 @@ class Lbann(CMakePackage):
     depends_on('cudnn', when='+gpu')
     depends_on('cub', when='+gpu')
     depends_on('mpi')
-    depends_on('hwloc')
-    depends_on('opencv@3.2.0: +openmp +core +highgui +imgproc +jpeg +png +tiff +zlib ~eigen', when='+opencv')
+    depends_on('hwloc ~pci ~libxml2')
+    # LBANN wraps OpenCV calls in OpenMP parallel loops, build without OpenMP
+    depends_on('opencv@3.2.0: +core +highgui +imgproc +jpeg +png +tiff +zlib '
+               '+fast-math ~lapack ~calib3d ~dnn ~features2d ~flann ~java ~ml '
+               '~python ~stitching ~superres ~ts ~video ~videostab ~videoio '
+               '~eigen ~pthreads_pf ~opencl ~opencl_svm ~openclamdfft '
+               '~openclamdblas', when='+opencv')
```

Comment:
Just to confirm, every one of these variants is absolutely required to build `lbann`? A lot of people use variants that they "prefer", but aren't actually required. We're trying to cut down on that as it restricts what combinations people can build.